### ISSUE TITLE
Ignore SPAM keyword in subject header

### ISSUE THREAD

**Issue by acorey0** [lieser/dkim_verifier#202]
My ISP's spam scoring system alters high-scoring emails by adding the score to the Subject header. For example:
`Subject: ****SPAM:13.5**** Kate Winslet’s Latest Investment Has The Government And Big Banks Terrified.`

I don't know how common a scenario this is, but if DKIM validation fails I would like to know whether it would pass without the `****SPAM:13.5**** `. Because the inserted string is not fixed, I would probably have to specify it using a regex or similar.

**Comment by lieser** [comment]
I also don't know how common this is. Note that unless there will be more people requesting something like this, the priority to implement this will be rather low.

Have you looked if your server makes a DKIM verification, and saves the result in the `Authentication-Results` header? In that case enabling [the reading of the ARH header](https://github.com/lieser/dkim_verifier/wiki/Options#read-authentication-results-header) may already achieve what you want.

**Comment by acorey0** [comment]
Thanks for the suggestion. They don't add Authentication-Results, but that's another enhancement I could suggest.

**Comment by ohbobva** [comment]
Yes, it would be nice to optionally ignore the SPAM tag in the email subject.  In my case it's "**SPAM**".

**Comment by D0LLYNH0** [comment]
I’m not particularly interested in this feature being implemented, just adding some info in case it happens in the future...

Some patterns I’ve seen out there:

```
[SPAM]
** SPAM **
***SPAM***
*** SPAM ***
```

A possible regex to cover these variations:

```regex
^[* []+SPAM[] *]+
```

But there’s also the pattern @acorey0 mentioned... well, this will scale a bit over time...

It’s relatively easy to find raw emails online to collect real-world patterns.
